<a href="https://colab.research.google.com/github/owaiskhan786/PIAIC_BATCH36_PythonProgram/blob/main/PracticeAssignmentClass6.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### Task1:
You need to code a program that prints numbers from 1 to 20 except multiple of 2 & 3.

Hint: use "modulo" operator and "and/or" operators

In [ ]:
for i in range(1,20):
  if i % 3 == 0 or i % 2 == 0:
        continue
  else:
    print(i)

1
5
7
11
13
17
19


### Task2:
Write a program that keep on accepting number from the user until user enters Zero. Display the sum and average of all the numbers.

Hint: Use while loop with flag. 

In [ ]:
count = 0
sum = 0.0
number = 1

while number != 0:
	number = int(input("Enter Number :"))
	sum = sum + number
	count += 1

if count == 0:
	print("Input some numbers")
else:
	print("Average and Sum of the above numbers are: ", sum / (count-1), sum)

Enter Number :2
Enter Number :6
Enter Number :4
Enter Number :8
Enter Number :9
Enter Number :0
Average and Sum of the above numbers are:  5.8 29.0


### Task3:
Accept 10 numbers from the user and display their average.

Hint: Use loop for accepting numbers or use split to take input in one go

In [ ]:
count = 0
sum = 0.0
for i in range(0,10):
  number = int(input("Enter Number : "))
  sum = sum + number
  count += 1

if count == 0:
	print("Input some numbers")
else:
	print("Average and Sum of the above numbers are: ", sum / (count-1), sum)

Enter Number : 1
Enter Number : 2
Enter Number : 3
Enter Number : 4
Enter Number : 5
Enter Number : 6
Enter Number : 7
Enter Number : 8
Enter Number : 9
Enter Number : 10
Average and Sum of the above numbers are:  6.111111111111111 55.0


### Task4:
Write a program to display sum of odd numbers and even numbers that fall between 12 and 37(including both numbers)


In [ ]:
even_total = 0
odd_total= 0
for number in range(12, 37):
    if(number % 2 == 0):
        even_total = even_total + number
    else:
        odd_total = odd_total + number
 
print("The Sum of Even Numbers from 12 to 37 = {1}".format(number, even_total))
print("The Sum of Odd Numbers from 12 to 37 = {1}".format(number, odd_total))

The Sum of Even Numbers from 12 to 37 = 312
The Sum of Odd Numbers from 12 to 37 = 288


### Task5:
Write a program to print table of a number accepted from user.


In [ ]:
Number = int(input("Enter Number"))
for i in range(1,11):
  print(f"{Number} x {i} = {Number * i}")


Enter Number7
7 x 1 = 7
7 x 2 = 14
7 x 3 = 21
7 x 4 = 28
7 x 5 = 35
7 x 6 = 42
7 x 7 = 49
7 x 8 = 56
7 x 9 = 63
7 x 10 = 70


### Task6:
Write a program to display product of the digits of a number accepted from the user.

If user inputs:123 Outputs 6

If user inputs: 567 Outputs 210

In [ ]:
Task_number = input("Enter Number : ")
sum = 0
for i in Task_number:
    sum = sum + int(i)
print(f" Sum : {sum}")

Enter Number : 589
 Sum : 22


### Task7: 
Write a program to print first 10 even numbers in reverse order.

In [ ]:
arr = []
for i in range(1,20):
  if i%2 == 0:
    arr.append(i)
    print(arr)
arr.reverse()

print(arr)

[2]
[2, 4]
[2, 4, 6]
[2, 4, 6, 8]
[2, 4, 6, 8, 10]
[2, 4, 6, 8, 10, 12]
[2, 4, 6, 8, 10, 12, 14]
[2, 4, 6, 8, 10, 12, 14, 16]
[2, 4, 6, 8, 10, 12, 14, 16, 18]
[18, 16, 14, 12, 10, 8, 6, 4, 2]


### Task8:
Write a program to find the sum of the digits of a number accepted from user

If user inputs:123 Outputs 6

If user inputs: 567 Outputs 18

In [ ]:
Task_number = input("Enter Number : ")
sum = 0
for i in Task_number:
    sum = sum + int(i)
print(f" Sum : {sum}")

Enter Number : 567
 Sum : 18


### Task9:
Write a program that accepts a sentence and calculate the number of letters and digits.

Suppose the following input is supplied to the program:

hello world! 123

Then, the output should be:

LETTERS 10

DIGITS 3


In [ ]:
Number_Count = 0
Letter_Count = 0
Task_sentence = input("Enter Sentence : ")
for i in Task_sentence:
  if i.isdigit():
    Number_Count += 1
  else:
    Letter_Count += 1
print(f"Letter : {Letter_Count}")
print(f"Digit : {Number_Count}")

Enter Sentence : hello world! 123
Letter : 13
Digit : 3


### Task10: 
You need to trace and make a list of all such numbers from1000 to 3000 in which all the digits are even numbers.

In [38]:
arr = []
for i in range(1000,3000):
  if i%2 == 0:
    arr.append(i)
print(arr)

[1000, 1002, 1004, 1006, 1008, 1010, 1012, 1014, 1016, 1018, 1020, 1022, 1024, 1026, 1028, 1030, 1032, 1034, 1036, 1038, 1040, 1042, 1044, 1046, 1048, 1050, 1052, 1054, 1056, 1058, 1060, 1062, 1064, 1066, 1068, 1070, 1072, 1074, 1076, 1078, 1080, 1082, 1084, 1086, 1088, 1090, 1092, 1094, 1096, 1098, 1100, 1102, 1104, 1106, 1108, 1110, 1112, 1114, 1116, 1118, 1120, 1122, 1124, 1126, 1128, 1130, 1132, 1134, 1136, 1138, 1140, 1142, 1144, 1146, 1148, 1150, 1152, 1154, 1156, 1158, 1160, 1162, 1164, 1166, 1168, 1170, 1172, 1174, 1176, 1178, 1180, 1182, 1184, 1186, 1188, 1190, 1192, 1194, 1196, 1198, 1200, 1202, 1204, 1206, 1208, 1210, 1212, 1214, 1216, 1218, 1220, 1222, 1224, 1226, 1228, 1230, 1232, 1234, 1236, 1238, 1240, 1242, 1244, 1246, 1248, 1250, 1252, 1254, 1256, 1258, 1260, 1262, 1264, 1266, 1268, 1270, 1272, 1274, 1276, 1278, 1280, 1282, 1284, 1286, 1288, 1290, 1292, 1294, 1296, 1298, 1300, 1302, 1304, 1306, 1308, 1310, 1312, 1314, 1316, 1318, 1320, 1322, 1324, 1326, 1328, 1330, 133

### Task11: 
You need to write a program that counts the number of vowels in a given sentence as input from console. 

Suppose the following input is supplied to the program:

Hello world! Practice makes perfect

Then, the output should be:

a appeared 2 times

e appeared 5 times

i appeared  1 time

o appeared 2 times

u appeared 0 time


In [9]:
string=input("Enter string:")
vowels=0
for i in string:
      if(i=='a' or i=='e' or i=='i' or i=='o' or i=='u' or i=='A' or i=='E' or i=='I' or i=='O' or i=='U'):
            vowels=vowels+1
print("Number of vowels are:")
print(vowels)

Enter string:Hello world! Practice makes perfect
Number of vowels are:
10


### Task12:
You need to write a program that takes sequence of lines as input and prints the lines after making all characters in the sentence capitalized.

Suppose the following input is supplied to the program:

Hello world

Practice makes perfect

Then, the output should be:

HELLO WORLD
PRACTICE MAKES PERFECT

In [4]:
lines = []
while True:
    l = input()
    if l:
        lines.append(l.upper())
    else:
        break;

for l in lines:
    print(l)

Hello world  Practice makes perfect

HELLO WORLD  PRACTICE MAKES PERFECT


### Task13:
You need to write a code that takes a comma separated sequence of words as input and prints the words in a comma-separated sequence after sorting them alphabetically.

Suppose the following input is supplied to the program:

without,hello,bag,world

Then, the output should be:

bag,hello,without,world

In [5]:
items = input("Input comma separated sequence of words")
words = [word for word in items.split(",")]
print(",".join(sorted(list(set(words)))))

Input comma separated sequence of wordswithout,hello,bag,world
bag,hello,without,world


### Task14:
You need code a program that calculates and prints the value according to the given formula:
P = Square root of [(2 * A * B)/C]

Following are the fixed values of A and B:

A is 50. B is 30.

The values of the literal C should be taken as console input to your program in a comma-separated sequence.

Example:
Let us assume the following comma separated input sequence is given to the program:
100,150,180

The output of the program should be:
18,22,24

In [6]:
import math
c=50
h=30
value = []
items=[x for x in input().split(',')]
for d in items:
    value.append(str(int(round(math.sqrt(2*c*float(d)/h)))))

print(','.join(value))

100,150,180
18,22,24


### Task15:
You need to write a ocde in python that finds all such numbers which are divisible by 7 but are not a multiple of 5,
between 2000 and 3200 (both included).

Print the obtained numbers in a comma-separated sequence on a single line

Hint: modulo operator, and/or operator, for loop /while loop

In [ ]:
# Your code goes here

### Task16:
With a given integral number n, write a program to generate a dictionary that contains (i, i*i) such that is an integral number between 1 and n (both included). and then the program should print the dictionary.

Suppose the following input is supplied to the program:

8

Then, the output should be:

{1: 1, 2: 4, 3: 9, 4: 16, 5: 25, 6: 36, 7: 49, 8: 64}

In [7]:
number = int(input("Type a number: "))

numberDict = {}
for i in range(1, number+1):
    numberDict[i] = i*i

print(numberDict)

Type a number: 8
{1: 1, 2: 4, 3: 9, 4: 16, 5: 25, 6: 36, 7: 49, 8: 64}


### Task17:
You need to write a code which accepts a sequence of comma separated 4 digit binary numbers as its input and then check whether they are divisible by 5 or not. 

The numbers that are divisible by 5 are to be printed in a comma separated sequence.

Example:

0100,0011,1010,1001

Then the output should be:
1010

Reference: 
1) https://www.datacamp.com/.../python-data-type-conversion

2) https://stackoverflow.com/questions/21765779/converting-binary-to-decimal-integer-output?fbclid=IwAR3DD_oiq5pn99IjyxSJgojvB5W6M4JqQtv4EUl_IF_DtmuRXlP-DswcKAA

In [8]:
items = []
num = [x for x in input().split(',')]
for p in num:
    x = int(p, 2)
    if not x%5:
        items.append(p)
print(','.join(items))

0100,0011,1010,1001
1010
